Go CLI Fails Returns segmentation violation When Cannot Connect

Stacktrace: 

```bash
$ ./zally lint ../../server/src/test/resources/fixtures/api_spp.json 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x12a2d9f]

goroutine 1 [running]:
github.com/zalando-incubator/zally/cli-go/zally/domain.(*Violations).filterViolations(0x0, 0x134a7bd, 0x4, 0x131ef00, 0x1, 0xc42017c070)
	/Users/netme/w/go/src/github.com/zalando-incubator/zally/cli-go/zally/domain/violations.go:54 +0x4f
github.com/zalando-incubator/zally/cli-go/zally/domain.(*Violations).Must(0x0, 0xc42017c070, 0x0, 0x0)
	/Users/netme/w/go/src/github.com/zalando-incubator/zally/cli-go/zally/domain/violations.go:34 +0x40
github.com/zalando-incubator/zally/cli-go/zally/domain.(*Violations).ToString(0x0, 0xc4200ec000, 0xb297)
	/Users/netme/w/go/src/github.com/zalando-incubator/zally/cli-go/zally/domain/violations.go:19 +0x5a
github.com/zalando-incubator/zally/cli-go/zally/commands.lintFile(0x7fff5fbff8e5, 0x35, 0xc42004b8e0, 0x0, 0x0)
	/Users/netme/w/go/src/github.com/zalando-incubator/zally/cli-go/zally/commands/lint.go:47 +0xb6
github.com/zalando-incubator/zally/cli-go/zally/commands.lint(0xc4200d4280, 0x0, 0xc4200d4280)
	/Users/netme/w/go/src/github.com/zalando-incubator/zally/cli-go/zally/commands/lint.go:37 +0x118
github.com/urfave/cli.HandleAction(0x12e6a80, 0x135c8e0, 0xc4200d4280, 0xc42006c300, 0x0)
	/Users/netme/w/go/src/github.com/urfave/cli/app.go:485 +0xd4
github.com/urfave/cli.Command.Run(0x134a8e5, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x135585f, 0x25, 0x0, ...)
	/Users/netme/w/go/src/github.com/urfave/cli/command.go:207 +0xb6e
github.com/urfave/cli.(*App).Run(0xc4200dc1a0, 0xc420010240, 0x3, 0x3, 0x0, 0x0)
	/Users/netme/w/go/src/github.com/urfave/cli/app.go:250 +0x7d0
main.main()
	/Users/netme/w/go/src/github.com/zalando-incubator/zally/cli-go/zally/main.go:11 +0x57
```

--- a/cli-go/zally/commands/lint.go
+++ b/cli-go/zally/commands/lint.go
@@ -44,9 +44,12 @@ func lintFile(path string, requestBuilder *utils.RequestBuilder) error {
 	}
 
 	violations, err := doRequest(requestBuilder, data)
-	fmt.Print(violations.ToString())
+	if err != nil {
+		return err
+	}
 
-	return err
+	fmt.Print(violations.ToString())
+	return nil
 }
 
 func readFile(path string) (json.RawMessage, error) {

